src/team_lib/ucx/allgather/allgather_ring.c

```diff
@@ -64,7 +64,7 @@ xccl_status_t xccl_ucx_allgather_ring_start(xccl_ucx_collreq_t *req)
     ptrdiff_t sbuf       = (ptrdiff_t)req->args.buffer_info.src_buffer;
     ptrdiff_t rbuf       = (ptrdiff_t)req->args.buffer_info.dst_buffer;
 
-    if (sbuf != rbuf) {
+    if (sbuf != rbuf || group_rank) {
```

Comment:
@Sergei-Lebedev @bureddy we have a problem there i believe. We incorrectly defined INPLACE in xccl in general, assuming sbuf==rbuf is inplace case. IT is not. For example, for the allgather case the INPLACE option means that there is only 1 buffer to get the result of the operation and each process already has its local send data in the proper location of the buffer : rbuf + data_size*group_rank. The problem with xccl is that we haven't defined at all in xccl.h. Is it sbuf == rbuf or is it sbuf == rbuf+data_size*group_rank. The former is incorrect for allgather. The latter is incorrect for other ops (say allreduce). DEfining a special flag would be a proper way imho.